pkg/controllers/dynakube/controller.go

```diff
@@ -261,14 +261,14 @@ func (controller *Controller) reconcileDynaKube(ctx context.Context, dynakube *d
 
 	log.Info("start reconciling process module config")
 
-	err = processmoduleconfigsecret.NewReconciler(
-		controller.client, controller.apiReader, dynatraceClient, dynakube, controller.scheme, timeprovider.New()).
-		Reconcile(ctx)
+	err = controller.reconcileComponents(ctx, dynatraceClient, istioClient, dynakube)
 	if err != nil {
 		return err
 	}
 
-	return controller.reconcileComponents(ctx, dynatraceClient, istioClient, dynakube)
+	return processmoduleconfigsecret.NewReconciler(
```

Comment:
why here?

ProcessModuleConfig only matters if there is injection (cloudnative/appmon), so it could be in the [injection reconciler](https://github.com/Dynatrace/dynatrace-operator/blob/main/pkg/controllers/dynakube/injection/reconciler.go#L62-L91)

Reply:
It used to be here and for the refactoring to work I needed to move reconcilers because the oneagent tenant token needs to be created in order to set oneagent connection info. But I will have a look if I can move it into the injection reconciler.

Reply:
done in https://github.com/Dynatrace/dynatrace-operator/pull/2777/commits/21e52638bf7c80d7a47af88adb0f0875282826bc